snow/engine/avalanche/vertex/stateless_vertex_test.go

```diff
@@ -0,0 +1,138 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package vertex
+
+import (
+	"testing"
+
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+func TestVertexVerify(t *testing.T) {
+	tests := []struct {
+		name      string
+		vertex    StatelessVertex
+		shouldErr bool
+	}{
+		{
+			name:      "zero vertex",
+			vertex:    statelessVertex{innerStatelessVertex: innerStatelessVertex{}},
+			shouldErr: true,
+		},
+		{
+			name: "valid vertex",
+			vertex: statelessVertex{innerStatelessVertex: innerStatelessVertex{
+				Version:      0,
+				ChainID:      ids.ID{},
+				Height:       0,
+				Epoch:        0,
+				ParentIDs:    []ids.ID{},
+				Txs:          [][]byte{{}},
+				Restrictions: []ids.ID{},
+			}},
+			shouldErr: false,
+		},
+		{
+			name: "invalid vertex epoch",
+			vertex: statelessVertex{innerStatelessVertex: innerStatelessVertex{
+				Version:      0,
+				ChainID:      ids.ID{},
+				Height:       0,
+				Epoch:        1,
+				ParentIDs:    []ids.ID{},
+				Txs:          [][]byte{{}},
+				Restrictions: []ids.ID{},
+			}},
+			shouldErr: true,
+		},
+		{
+			name: "too many vertex parents",
+			vertex: statelessVertex{innerStatelessVertex: innerStatelessVertex{
+				Version:      0,
+				ChainID:      ids.ID{},
+				Height:       0,
+				Epoch:        0,
+				ParentIDs:    make([]ids.ID, maxNumParents+1),
```

Comment:
Wouldn't this test error due to the parent IDs not being unique even if the max num parents wasn't enforced?

Reply:
Updated